### PR TITLE
feat(FO-2205): add new images for `r` and `jupyter`

### PR DESCRIPTION
**Pull request by peak-jiteshahuja**
## Description

Description of what this PR does. What have you added or changed, and why?  If it fixes a bug or resolves a feature request, be sure to link to that issue.

This PR aims to add the following new images:

1. `workspace-python-ds-pack-2.1.0-base-python-3.8.17`
2. `workspace-python-ds-pack-2.1.0-base-python-3.9.18`
3. `workspace-python-ds-pack-2.1.0-base-python-3.10.13`
4. `workspace-python-ds-pack-2.1.0-base-python-3.11.5`
5. `workspace-r-ds-pack-1.1.0-base-r-4.3.1`

More details cab be found in the [JIRA ticket](https://peak-bi.atlassian.net/browse/FO-2205).

## Review Checks

Please check if the PR fulfills these requirements:

_Put an `x` in the boxes that apply, Remove any lines that do not apply_
- [x] 📝 The commit message is clear and descriptive
- [x] 🔐 The Security Considerations section in the PR description is complete - **Please do not remove this**
- [ ] ✅ Tests for the changes have been added and run successfully including the new changes
- [x] 📄 Documentation has been added / updated (for bug fixes / features)

## Dependencies

Add links to any pull requests or documentation related to this pull request.

N/A

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
- [ ] Yes
- [x] No

## Security Considerations

Are there any other security or data concerns to be aware of?

Please discuss the security implications/considerations relevant to the proposed change. 
This may include...
* security-relevant design decisions
* concerns 
* important discussions
* implementation-specific guidance and pitfalls
* an outline of threats and risks and how they are being addressed.

1. no critical/high severity snyk vulnerabilities should remain

## Types of change

What kind of change does this Pull Request introduce?
_Put an `x` in the boxes that apply_

- [ ] 🐛 Bugfix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 Adding or updating configuration files, development scripts etc.
- [ ] ♻️ Refactoring (no functional changes, no api changes)
- [ ] 🧹 Chore (removing redunant files, fixing typos etc.)
- [ ] 📄 Documentation Update 
- [ ] ❓ Other (if none of the other choices apply)

If "Other" please specify

**YOUR TEXT HERE**

# Testing

Please include steps that the reviewer can follow in order to test the changes

Verify if the new global images are working fine.

# Other information

N/A

### CODE REVIEW

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+# install system packages
+RUN apt-get update \
+    # installation requirements
+    && apt-get install -y --no-install-recommends \
+    apt-transport-https \
+    ca-certificates \
+    software-properties-common \
+    curl \
+    gnupg2 \
+    # docker
```
Comment (akshayjain04) [inline workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile:15]: Hey @peak-jiteshahuja, can we remove this comment

**Review by akshayjain04 — COMMENTED** [review]
Hey @peak-jiteshahuja, can we add the snyk report screenshot in the PR

**Review by akshayjain04 — COMMENTED** [review]
Hey @peak-jiteshahuja, the vulnerabilities doesn't seem to have reduced. Am I missing something 🤔 

Screenshot 2023-09-20 at 2 11 04 PM

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# workspace-python-ds-pack-2.1.0-base-python-3.10.13
+This is the Python Data Science pack image made for Workspaces. 
+The image runs a [Jupyterhub](https://jupyter.org/hub) server that has [PyCharm](https://lp.jetbrains.com/projector/) and [VSCode](https://github.com/coder/code-server) pre-installed with all of the essential and most used packages.
+The images also comes pre installed with [peak-sdk](https://docs.peak.ai/sdk/).
```
Comment (akshayjain04) [inline workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md:4]:
```suggestion
The images also comes pre installed with [peak-sdk:1.0.0](https://docs.peak.ai/sdk/).
```
NIT

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+### VS Code extensions installed
+```
+cweijan.vscode-database-client2   6.6.3    
+```
+
+## Building the image
+To build the image locally, run the docker build command and pass in the required build arguments:
+```
+docker build . -t workflow-python-ds-pack-2.1.0-base-python-3.8.17
```
Comment (akshayjain04) [inline workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md:69]:
```suggestion
docker build . -t workflow-python-ds-pack-2.1.0-base-python-3.10.13
```

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+```
+aws-cli        2.7.4
+curl           7.88.1
+git            2.39.2
+jq             1.6
+nano           7.2
+vim            9.0
+fish           3.6.0
+zsh            5.9
+R              4.2.0
```
Comment (akshayjain04) [inline workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md:27]: I think we can remove `R`

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/codeserver-proxy/jupyter_codeserver_proxy/__init__.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+def _install_vscode_extension():
+  # Run the Bash command and capture the output
+  try:
+      result = subprocess.check_output(bash_command, shell=True, text=True)
+      print("Bash command output:")
+      print(result)
+  except subprocess.CalledProcessError as e:
+      print("Error running the Bash command:")
+      print(e)
```
Comment (akshayjain04) [inline workspace/python/ds-pack/2.1.0-base-python-3.10.13/codeserver-proxy/jupyter_codeserver_proxy/__init__.py:28]: NIT: newline

File: workspace/python/ds-pack/2.1.0-base-python-3.11.5/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+python-lsp-server[all]            1.4.1
+virtualenv                        20.24.5
+peak-sdk                          1.0.0
+jupysql                           0.10.1
+```
+
+## Building the image
+To build the image locally, run the docker build command and pass in the required build arguments:
+```
+docker build . -t workflow-python-ds-pack-2.1.0-base-python-3.8.17
```
Comment (akshayjain04) [inline workspace/python/ds-pack/2.1.0-base-python-3.11.5/README.md:69]:
```suggestion
docker build . -t workflow-python-ds-pack-2.1.0-base-python-3.11.5
```

File: workspace/python/ds-pack/2.1.0-base-python-3.11.5/codeserver-proxy/jupyter_codeserver_proxy/__init__.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+def _install_vscode_extension():
+  # Run the Bash command and capture the output
+  try:
+      result = subprocess.check_output(bash_command, shell=True, text=True)
+      print("Bash command output:")
+      print(result)
+  except subprocess.CalledProcessError as e:
+      print("Error running the Bash command:")
+      print(e)
```
Comment (akshayjain04) [inline workspace/python/ds-pack/2.1.0-base-python-3.11.5/codeserver-proxy/jupyter_codeserver_proxy/__init__.py:28]: NIT: newline

File: workspace/README.md [hunk 8]
```diff
@@ -5,7 +5,13 @@ Write and develop solutions in RStudio and Jupyter using Peak workspaces.
 We provide Python and R versions of the images:
 - [Python](./python)
   - [ds-pack](./python/ds-pack)
-    - [workspace-python-ds-pack-v1](./python/ds-pack/v1/)
+    - [workspace-python-ds-pack-1.0.0-base-python-3.8.13](./python/ds-pack/1.0.0-base-python-3.8.13/)
+    - [workspace-python-ds-pack-2.0.0-base-python-3.11.2](./python/ds-pack/2.0.0-base-python-3.11.2/)
+    - [workspace-python-ds-pack-2.1.0-base-python-3.8.17](./python/ds-pack/2.1.0-base-python-3.8.18/)
+    -  [workspace-python-ds-pack-2.1.0-base-python-3.9.18](./python/ds-pack/2.1.0-base-python-3.9.18/)
```
Comment (saurabh0402) [inline workspace/README.md:11]: Extra space

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
+    # install system packages
+    # jq is needed for the Peak Platform
+    jq \
+    nano \
+    vim \
+    fish \
+    zsh \
+    # r-base \
```
[the remark this answers is not in the document]
Comment (saurabh0402) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile:34]: Any reason for removing this? Don't think we should remove this.

File: workspace/python/ds-pack/2.1.0-base-python-3.8.18/README.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+### VS Code extensions installed
+```
+cweijan.vscode-database-client2   6.6.3    
+```
+
+## Building the image
+To build the image locally, run the docker build command and pass in the required build arguments:
+```
+docker build . -t workflow-python-ds-pack-2.1.0-base-python-3.8.17
```
Comment (akshayjain04) [inline workspace/python/ds-pack/2.1.0-base-python-3.8.18/README.md:69]:
```suggestion
docker build . -t workflow-python-ds-pack-2.1.0-base-python-3.8.18
```

File: workspace/python/ds-pack/2.1.0-base-python-3.8.18/codeserver-proxy/jupyter_codeserver_proxy/__init__.py [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+def _install_vscode_extension():
+  # Run the Bash command and capture the output
+  try:
+      result = subprocess.check_output(bash_command, shell=True, text=True)
+      print("Bash command output:")
+      print(result)
+  except subprocess.CalledProcessError as e:
+      print("Error running the Bash command:")
+      print(e)
```
Comment (akshayjain04) [inline workspace/python/ds-pack/2.1.0-base-python-3.8.18/codeserver-proxy/jupyter_codeserver_proxy/__init__.py:28]: NIT: newline

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    && pip install --no-cache-dir ./projector-proxy
+
+# force shell to bash as Peak Platform isn't respecting jupyterhub option
+RUN ln -sf /bin/bash /bin/sh
+
+EXPOSE 8000
+CMD ["jupyterhub", \
+    "--Spawner.default_url=/lab", \
+    "--Spawner.http_timeout=90", \
+    "--NotebookApp.terminado_settings={'shell_command':'[/bin/bash]'}" \
```
Comment (saurabh0402) [inline workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile:107]: This has changed. Any reason for this? 🤔

File: workspace/r/ds-pack/1.1.0-base-r-4.3.1/Dockerfile
Comment (akshayjain04) [inline workspace/r/ds-pack/1.1.0-base-r-4.3.1/Dockerfile:1]: Any reason for `1.1.0`. If its a new image, should we make `2.0.0` or similar

File: workspace/r/ds-pack/1.1.0-base-r-4.3.1/README.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+### ODBC Setup
+- The `odbcinst.ini` file exists in the `/etc` directory, and the `ODBCSYSINI` env has been set accordingly.
+
+You can find more details about setting up ODBC for `redshift` [here](https://docs.aws.amazon.com/redshift/latest/mgmt/configure-odbc-connection.html). For `snowflake` more details can be found [here](https://docs.snowflake.com/en/user-guide/odbc-linux.html)
+
+
+## Building the image
+To build the image locally, run the docker build command and pass in the required build arguments:
+```
+docker build . -t workflow-python-ds-pack-v1
```
Comment (akshayjain04) [inline workspace/r/ds-pack/1.1.0-base-r-4.3.1/README.md:57]: the image is changed right?

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/codeserver-proxy/jupyter_codeserver_proxy/__init__.py [hunk 14]
```diff
@@ -0,0 +1,28 @@
+import os
+import subprocess
+
+bash_command = "code-server --install-extension cweijan.vscode-database-client2@6.6.3"
+
+def setup_codeserver():
+  _install_vscode_extension()
```
Comment (saurabh0402) [inline workspace/python/ds-pack/2.1.0-base-python-3.10.13/codeserver-proxy/jupyter_codeserver_proxy/__init__.py:7]: Any reason for installing it here? We can install it from `Dockerfile` as well just like other Python packages?

File: workspace/python/ds-pack/2.1.0-base-python-3.11.5/Dockerfile [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+
+RUN pip install --no-cache-dir --upgrade \
+    pip \
+    setuptools \
+    wheel
+
+RUN pip install -r /tmp/requirements.txt
+
+# SPARK: copy over spark config
+COPY ./spark /usr/local/spark
```
Comment (saurabh0402) [inline workspace/python/ds-pack/2.1.0-base-python-3.11.5/Dockerfile:81]: Think we should remove this - [Ref](https://peak-ai.slack.com/archives/C02KUV4BVNF/p1695205524779779?thread_ts=1695205269.029099&cid=C02KUV4BVNF).

File: workspace/python/ds-pack/2.1.0-base-python-3.11.5/Dockerfile [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+
+# VS Code
+COPY codeserver-proxy codeserver-proxy
+RUN curl -fsSL https://code-server.dev/install.sh \
+    | sh -s -- --version 4.16.1 \
+    && pip install --no-cache-dir ./codeserver-proxy
+
```
Comment (saurabh0402) [inline workspace/python/ds-pack/2.1.0-base-python-3.11.5/Dockerfile:88]: We don't have Projector Proxy installed in this image?
Can we mention that somewhere if not already done.

**Review by saurabh0402 — COMMENTED** [review]
Tried out `workspace/r/ds-pack/1.1.0-base-r-4.3.1` and `workspace/python/ds-pack/2.1.0-base-python-3.10.13`. Looks fine.

**Review by akshayjain04 — COMMENTED** [review]
Tried workspaces with all 4 python images (present on olympus dev). Working fine there.

File: workspace/python/ds-pack/2.1.0-base-python-3.11.5/Dockerfile (hunk 16 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.11.5/Dockerfile:88]: We've actually removed it from readme from this image. Do we want to explicitly mention that it isn't present ?

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/codeserver-proxy/jupyter_codeserver_proxy/__init__.py (hunk 14 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/codeserver-proxy/jupyter_codeserver_proxy/__init__.py:7]: Hey yes. It actually doesn't work when we install it in the `Dockerfile` 😅

File: workspace/r/ds-pack/1.1.0-base-r-4.3.1/Dockerfile
Comment (peak-jiteshahuja) [reply workspace/r/ds-pack/1.1.0-base-r-4.3.1/Dockerfile:1]: Hey as discussed in the standup, 1.1.0 is more suitable for this image.

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile (hunk 12 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile:107]: Hey yes. The old format was actually not working and causing some issues :).

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+node           18.17.1
```
[the remark this answers is not in the document]
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md:28]: Il check and update accordingly 👍 .

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md (hunk 4 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md:27]: We will be keeping R.

**Comment by peak-jiteshahuja** [comment]
> Hey @peak-jiteshahuja, the vulnerabilities doesn't seem to have reduced. Am I missing something 🤔
> 
> EDIT: Okay got it, they are for old images. But there seems to be some high vulnerability in `workspace/python/ds-pack/2.0.0-base-python-3.11.2` image. Can we check those if they can be resolved
> 
> Screenshot 2023-09-20 at 2 11 04 PM

Hey we aren't updating any existing images, so we wouldn't be fixing any issues with `2.0.0-base-python-3.11.2`.

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile (hunk 9 above)
[the remark this answers is not in the document]
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile:34]: This has been done my mistake. I'l get it reverted.

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md (hunk 2 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md:4]: Hey i've actually mentioned the version in the readme so don't think we should be adding it here.

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md:28]: Hey i've verified it with a workspace. We are good with `18.17.1`.

File: workspace/r/ds-pack/1.1.0-base-r-4.3.1/README.md (hunk 13 above)
Comment (peak-jiteshahuja) [reply workspace/r/ds-pack/1.1.0-base-r-4.3.1/README.md:57]: Done.

File: workspace/python/ds-pack/2.1.0-base-python-3.8.18/codeserver-proxy/jupyter_codeserver_proxy/__init__.py (hunk 11 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.8.18/codeserver-proxy/jupyter_codeserver_proxy/__init__.py]: Done.

File: workspace/python/ds-pack/2.1.0-base-python-3.8.18/README.md (hunk 10 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.8.18/README.md]: Done.

File: workspace/README.md (hunk 8 above)
Comment (peak-jiteshahuja) [reply workspace/README.md]: Done.

File: workspace/python/ds-pack/2.1.0-base-python-3.11.5/codeserver-proxy/jupyter_codeserver_proxy/__init__.py (hunk 7 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.11.5/codeserver-proxy/jupyter_codeserver_proxy/__init__.py]: Done.

File: workspace/python/ds-pack/2.1.0-base-python-3.11.5/README.md (hunk 6 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.11.5/README.md]: Done.

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/codeserver-proxy/jupyter_codeserver_proxy/__init__.py (hunk 5 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/codeserver-proxy/jupyter_codeserver_proxy/__init__.py]: Done.

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md (hunk 3 above)
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md]: Done.

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile (hunk 9 above)
[the remark this answers is not in the document]
Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/Dockerfile]: Done.

**Review by akshayjain04 — APPROVED** [review]

File: workspace/python/ds-pack/2.1.0-base-python-3.11.5/Dockerfile (hunk 16 above)
Comment (saurabh0402) [reply workspace/python/ds-pack/2.1.0-base-python-3.11.5/Dockerfile:88]: Yeah, I think we should

Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.11.5/Dockerfile:89]: Done.

**Review by akshayjain04 — APPROVED** [review]

File: workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md [hunk 18]
```diff
@@ -24,7 +24,7 @@ nano           7.2
 vim            9.0
 fish           3.6.0
 zsh            5.9
-R              4.2.0
+R              4.3.1
```
Comment (akshayjain04) [inline workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md:27]: Hey Jitesh, its `4.2.2` right?

Comment (peak-jiteshahuja) [reply workspace/python/ds-pack/2.1.0-base-python-3.10.13/README.md:27]: Hey no thats the version of `r-base` package 😅 . The version of `R` installed in `4.3.1` :) .

**Review by saurabh0402 — APPROVED** [review]

**Pull request merged**